CMSgov/cms-carts-seds
# Migrate Upload Component

### Description

This is the sixth component being updated as part of [CMDCT-3811](https://jiraent.cms.gov/browse/CMDCT-3811). I'm updating a few things:

- Remove connect in favor of using hooks
- Migrate mapDispatchToProps to use hooks instead

Note in the video that when I upload something in one, it reflects later in the second component. Its because the components share ids. Not a bug, don't worry!

https://github.com/user-attachments/assets/afdd71da-326b-4295-a6bf-e9b19d6836ec


### Related ticket(s)


CMDCT-3811

---
### How to test

Run the application
Play around with the upload components. You'll find the dotted around section 3 quite liberally. Feel free to pick one!
It would be good to also check AWS. You'll find the deployed instance is over in the mdct-carts-dev env. If you enter it through cloudtamer and navigate to s3, you can do a search for the bucket called uploads-update-uploadcomponent-attachments. Inside of there is where you'll find the files you upload/delete from the form.

---
### Pre-review checklist


---
### Pre-merge checklist


#### Review
